dspace-uclouvain/src/main/java/org/dspace/uclouvain/submissionMetadataGenerators/generators/DegreeMappersService.java

```diff
@@ -0,0 +1,113 @@
+package org.dspace.uclouvain.submissionMetadataGenerators.generators;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.uclouvain.core.GenericResponse;
+import org.dspace.uclouvain.submissionMetadataGenerators.generators.model.DegreeMapper;
+import org.dspace.uclouvain.submissionMetadataGenerators.generators.model.DegreeMappers;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Service to read the degree mappers JSON data file.
+ */
+public class DegreeMappersService {
+    
+    private static final Logger log = LogManager.getLogger();
+
+    private String absoluteFilePath;
+
+    private DegreeMappers degreeMappers;
+
+    private Long lastModified;
+
+    @Autowired
+    private String source = DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("dspace.dir");
+
+
+    DegreeMappersService(String filePath) {
+        this.absoluteFilePath = this.source + filePath;
+        this.loadFileMappers(this.absoluteFilePath);
+    }
+
+    /**
+     * Load the given file mapping configuration 
```

Comment:
Try to be explicit in the docstring; a human should not read the code to understand how it works.
Something like 
```
Load the given file mapping configuration if required.
The configuration will be reloaded only if file content changed from the last call of the service.
...
```